### PR TITLE
Fixes #34280 - Add `ip_include?` method to `IPAM::Base`

### PR DESCRIPTION
**Pull request by Marahin**
Related to: https://github.com/theforeman/foreman_discovery/pull/545

https://projects.theforeman.org/issues/34280

### CODE REVIEW

File: app/services/ipam/base.rb [hunk 1]
```diff
@@ -17,13 +17,14 @@ def initialize(opts = {})
       @mac = nil if @mac.try(:blank?)
       @errors = ActiveModel::Errors.new(self)
       @block_ip_minutes = opts.fetch(:block_ip_minutes, BLOCK_IP_MINUTES_DEFAULT)
+      @ip_addr = IPAddr.new("#{subnet.network}/#{subnet.mask}")
```
Comment (ekohl) [inline app/services/ipam/base.rb]: In the other call we also include `subnet.family` to force the right family.
```suggestion
      @ip_addr = IPAddr.new("#{subnet.network}/#{subnet.mask}", subnet.family)
```

File: test/unit/ipam_test.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 require 'test_helper'
 
 class IPAMTest < ActiveSupport::TestCase
+
+
```
Comment (ekohl) [inline test/unit/ipam_test.rb]: Nit: could you leave these lines out?
```suggestion
```

File: app/services/ipam/base.rb (hunk 1 above)
Comment (Marahin) [reply app/services/ipam/base.rb]: Done

**Comment by ekohl** [comment]
ok to test

**Review by lzap — APPROVED** [review]
LGTM

**Comment by lzap** [comment]
Thank you very much for your contribution, this is now merged.

**Pull request merged**